### ISSUE TITLE
[false-negative]IntLongMath can not detect the case int converted to long when campare int variable with long variable

### ISSUE THREAD

**Issue by wuchiuwong** [google/error-prone#1663]
### Description of the problem / feature request:

IntLongMath can not detect the case int converted to long when campare int variable with long variable.
This rule is implemented in error_prone_core-2.3.4-sources.jar!\com\google\errorprone\bugpatterns\IntLongMath.java，
When searching for errors, this rule detects whether return statements(<code>matchReturn</code>), variable assignment statements(<code>matchAssignment</code>), and variable definition statements(<code>matchVariable</code>) contain int convert into long,
Therefore, when two variables are compared (for example longVar == intVar), type conversion may occur, this situation is ignored.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
if (score >= level * level * 1000) {
if (System.currentTimeMillis() - t > seconds * 1000) {
assertEquals(size * size, block.getStones().size());
assertEquals(count * 17, sequence.getLiveValue());
```

### What version of Error Prone are you using?

2.3.4

###  Have you found anything relevant by searching the web?

no

**Comment by lowasser** [comment]
When would this case actually lead to a bug?  Equality checking an int and a long, for example, shouldn't add causes of errors.

**Comment by omeryounus** [comment]
Which are IDE are you using? There is some smart cast that would convert long to int or int to short depending on their runtime value.
